### PR TITLE
feat(Autosuggest): Add mobile backdrop

### PR DESCRIPTION
**Pull request by kosanna**

### CODE REVIEW

**Comment by markdalgleish** [comment]
Would you be able to make this fade in? It's quite a jarring transition at the moment.

**Comment by kosanna** [comment]
@markdalgleish added transition

**Comment by kosanna** [comment]
Good to go, will just confirm re backdrop colour with Tim

File: theme/mixins/backdrop.less [hunk 1]
```diff
@@ -0,0 +1,11 @@
+.backdrop() {
+  position: fixed;
+  z-index: 0;
```
Comment (michaeltaranto) [inline theme/mixins/backdrop.less]: Is this `z-index` required? If so should we add an entry into [our z-index hierarchy](https://github.com/seek-oss/seek-style-guide/blob/master/theme/layout/z-indexes.less)?

File: react/Autosuggest/Autosuggest.less [hunk 2]
```diff
@@ -65,3 +65,18 @@
...
+
+.containerOpen + .autosuggestBackdrop {
+  .smallDeviceOnly({
+    .backdrop()
+  });
+}
+
+.containerOpen .isLabelCoveredWithBackdrop {
+  display: block;
+  pointer-events: none;
```
Comment (markdalgleish) [inline react/Autosuggest/Autosuggest.less]: 👏

File: theme.less [hunk 3]
```diff
@@ -1,14 +1,18 @@
...
 @import (reference) "theme/mixins/link";
 @import (reference) "theme/mixins/type";
+
+@import (reference) "theme/palette/accessible-variants";
+@import (reference) "theme/palette/brand";
+@import (reference) "theme/palette/elements";
+@import (reference) "theme/palette/grays";
+@import (reference) "theme/palette/partners";
+
+@import (reference) "theme/type/type";
```
Comment (markdalgleish) [inline theme.less]: Was this change deliberate?

Comment (kosanna) [inline theme.less]: Sort and split into groups, sorry for being annoying with that

**Comment by kosanna** [comment]
Tim confirmed colors/opacity are ok 👍

**Comment by markdalgleish** [comment]
@kosanna @michaeltaranto What are your thoughts on portrait iPad? Should it have the backdrop like mobile, or go without like desktop?

**Comment by markdalgleish** [comment]
Also, do we want to show a transparent backdrop on desktop to intercept clicks?

**Comment by kosanna** [comment]
@markdalgleish it's a difult functionality of autosuggest so no need to introduce a transparent backdrop

**Comment by kosanna** [comment]
Re landscape tablet...my opinion we should treat just based on width (?) regardless of the device.

**Comment by markdalgleish** [comment]
@kosanna the default behaviour for react-autosuggest is to close suggestions when clicking outside, but the click still registers with whatever you click on. This is different to how our classifications list behaves, and in this PR it's now different to the behaviour on mobile.

**Comment by markdalgleish** [comment]
@kosanna agreed about width, but I feel like showing the backdrop on <=768px widths could be a reasonable exception.

**Comment by kosanna** [comment]
@markdalgleish I've changed classification to behave in a similar way (it's in a PR on chalice), will pull it into a separate one today perhaps if this one is merged

**Comment by kosanna** [comment]
Change backdrop to appear on tablets. As we'll need to use it for classifications as well, we need to have a mixin exposed too

**Comment by kosanna** [comment]
Need second approval on this one :)
